### PR TITLE
feat: new modern extends and GS device converters update

### PR DESCRIPTION
**Pull request by mrskycriper**
## TODO

- [x] add `iasZoneAlarm` modern extend
  - [x] add to definition generator with generic settings
- [x] add `iasWarning` modern extend
  - [x] add to definition generator
- [x] add `ignoreClusterReport` modern extend that accepts any cluster
- [x] add `battery` modern extend
- [x] separate GS devices from Heiman and move to `gs.ts`
- [x] update GS device names
- [x] update all GS devices to modern extend

### CODE REVIEW

File: src/lib/modernExtend.ts [hunk 1]
```diff
@@ -842,3 +856,57 @@ export function occupancy(args?: Partial<BinaryArgs>): ModernExtend {
...
+    const fromZigbee: Fz.Converter[] = [{
+        cluster: args.cluster.toString(),
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {},
+    }];
+
+    return {fromZigbee, isModernExtend: true};
+}
+
+export function iasGas(args: {variant: 'alarm_1' | 'alarm_2'}): ModernExtend {
```
Comment (Koenkk) [inline src/lib/modernExtend.ts:870]: I propose to name this `iasAlarm` and add an additional arg to specify the property (e.g. `contact`, `gas`)

Comment (mrskycriper) [reply src/lib/modernExtend.ts:870]: That will be a big extend, but sounds reasonable enough to encapsulate everything iasZone related

File: src/lib/modernExtend.ts [hunk 2]
```diff
@@ -754,6 +754,101 @@ export function batteryPercentage(args?: Partial<NumericArgs>) {
     });
 }
 
+export interface BatteryArgs {
+    voltageToPercentage?: string | {min: number, max: number}, dontDividePercentage?: boolean,
```
Comment (Koenkk) [inline src/lib/modernExtend.ts:758]: I see a `voltageToPercentage` option, but there is no place in this function where voltage reporting is setup.

File: src/lib/modernExtend.ts [hunk 3]
```diff
@@ -842,3 +937,162 @@ export function occupancy(args?: Partial<BinaryArgs>): ModernExtend {
...
+        }
+        args.zoneAttributes.map((attr) => {
+            if (attr !== 'alarm_1' && attr !== 'alarm_2') exposes.push(exposeList[attr]);
+        });
+    }
+
+    const type = [];
+    if (args.msgType?.includes('notification')) type.push('commandStatusChangeNotification');
+    if (args.msgType?.includes('report')) type.push('attributeReport', 'readResponse');
+    if (type.length === 0) type.push('commandStatusChangeNotification');
```
Comment (Koenkk) [inline src/lib/modernExtend.ts:1018]: Let's include all converters by default, I don't see any reason not to.

Comment (mrskycriper) [reply src/lib/modernExtend.ts]: Fair enough, even tho the are not zcl compliant command types here

File: src/lib/modernExtend.ts (hunk 2 above)
Comment (mrskycriper) [reply src/lib/modernExtend.ts:744]: Done

File: src/lib/modernExtend.ts [hunk 4]
```diff
@@ -754,6 +754,101 @@ export function batteryPercentage(args?: Partial<NumericArgs>) {
     });
 }
 
+export interface BatteryArgs {
+    voltageToPercentage?: string | {min: number, max: number}, dontDividePercentage?: boolean,
+    batteryPercentage?: boolean, batteryVoltage?: boolean, batteryAlarm?: boolean,
+    batteryPercentageReporting?: ReportingConfigWithoutAttribute,
+}
+export function battery(args?: BatteryArgs): ModernExtend {
```
[the remark this answers is not in the document]
Comment (mrskycriper) [reply src/lib/modernExtend.ts:749]: Removed and refactored all mentions to the new extend.

**Comment by mrskycriper** [comment]
@Koenkk resolved merge conflict, ready for merge now.

File: src/lib/modernExtend.ts [hunk 5]
```diff
@@ -750,18 +750,117 @@ export function co2(args?: Partial<NumericArgs>) {
...
+            return payload;
+        },
+    }];
+
+    const defaultReporting: ReportingConfigWithoutAttribute = {min: '1_HOUR', max: 'MAX', change: 10};
+
+    const configure: Configure = async (device, coordinatorEndpoint, logger) => {
+        if (args.percentageReporting) {
+            if (args.percentageReportingConfig) {
+                await setupAttributes(device, coordinatorEndpoint, 'genPowerCfg', [
```
Comment (Koenkk) [inline src/lib/modernExtend.ts:841]: Please use:

```js
await setupAttributes(device, coordinatorEndpoint, 'genPowerCfg', [
                    {attribute: 'batteryPercentageRemaining', ...(args.percentageReportingConfig ?? defaultReporting)},
                ], logger);
```

Same for voltage reporting below.

Comment (mrskycriper) [reply src/lib/modernExtend.ts]: Done.

**Comment by Koenkk** [comment]
Another great pr, thanks!

**Pull request merged**